cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs

```diff
@@ -108,3 +113,58 @@ fn xcm_prologue_fee() {
 	let buffered_fee = execution_fee * 2;
 	println!("buffered execution fee for prologue for 8 assets: {}", buffered_fee);
 }
+
+#[test]
+fn register_token_xcm() {
+	BridgeHubWestend::execute_with(|| {
+		let weth_contract = H160::zero(); // weth contract placeholder
+		let token: H160 = H160::zero(); // token id placeholder
+		let owner: H256 = H256::zero(); // bridge placeholder
+		let dot_to_eth_rate: u128 = 2_500_000_000_000_000;
+
+		let ethereum_network_v5: NetworkId = EthereumNetwork::get().into();
+		let weth_asset = Location::new(
+			2,
+			[
+				GlobalConsensus(ethereum_network_v5),
+				AccountKey20 { network: None, key: weth_contract.into() },
+			],
+		);
+		let dot_asset = Location::new(1, Here);
+
+		let weth_asset_creation: u128 = (CreateAssetDeposit::get() * dot_to_eth_rate) / 10_u128.pow(18);
+		let weth_fee: xcm::prelude::Asset = (weth_asset, weth_asset_creation).into();
+		let dot_fee: xcm::prelude::Asset = (dot_asset, CreateAssetDeposit::get()).into();
+
+		println!("weth_asset_creation: {:?}", weth_asset_creation);
+
+		let asset_id = Location::new(
+			2,
+			[GlobalConsensus(ethereum_network_v5), AccountKey20 { network: None, key: token.into() }],
+			);
+
+		let register_token_xcm =
+			vec![
+				ExchangeAsset { give: weth_fee.into(), want: dot_fee.clone().into(), maximal: false },
+				PayFees { asset: dot_fee },
```

Comment:
Should check with Cisco/Adrian, but on AH it should be possible to pay fees directly with WETH without having to exchange first.

I guess either way, you can test this using an xcm simulation test.

Reply:
Now that you mention it, I tested the automatic exchange from Weth to DOT with the unified rewards work - it works.

Register token xcm: 0508310100000700b2118417060315013500020209049edaa8020300**5861980c0268778f56f803753e3892a2fb341a53**00**e5eaf816d3384d366e1eed03a1f0bf67823068c95947dae4455b020216f0cf61**01000000

Replace token ID: 0x5861980c0268778f56f803753e3892a2fb341a53
Replace bridge owner ID: 0xe5eaf816d3384d366e1eed03a1f0bf67823068c95947dae4455b020216f0cf61

Reply:
Nice!

`owner` should be constructed the same as in v1 though:

```
let owner = GlobalConsensusEthereumConvertsFor::<[u8; 32]>::from_chain_id(&chain_id);
```

Reply:
That's a good point!

token: 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
Register token hex Sepolia: "0508310100000700b2118417060315013500020209049edaa8020300c**02aaa39b223fe8d0a0e5c4f27ead9083c756cc2**00ce796ae65569a670d0c1cc1ac12515a3ce21b5fbf729d63d7b289baad070139d01000000"
Register token Mainnet: 0508310100000700b2118417060315013500020209049edaa8020300c**02aaa39b223fe8d0a0e5c4f27ead9083c756cc2**00204dfe37731e8e2b4866ad0da9a17c49f434542c3477c5f914a3349acd88ba1a01000000

Reply:
Looks like you're still passing the dot fee though: `PayFees { asset: dot_fee },`. Shouldn't it be weth?

Reply:
Hmmm I don't know. We provide a weth fee, but the asset creation fee amount is in dot. So perhaps we should specify a dot fee here and the weth will be traded for dot? Otherwise we again need to keep some exchange rate to specify a sufficient weth amount here in `PayFees`. Correct me if I am wrong.

Reply:
Added ExchangeAsset back: https://github.com/claravanstaden/polkadot-sdk/pull/4/files#diff-8397e49665ab41c48eb0755a8a6b2510bbe94b0c00d808e6d167c3b0e2d0db26R160

Updated xcm for Sepolia:
050c0f0004020209049edaa8020300**be68fc2d8249eb60bfcf0e71d5a0d2f2e292c4ed**000f**00c06e31d91001**040100000700b211841700310100000700b2118417060315013500020209049edaa8020300**c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2**00ce796ae65569a670d0c1cc1ac12515a3ce21b5fbf729d63d7b289baad070139d01000000

We need to replace these three values:
- Weth Token ID: `0xbe68fc2d8249eb60bfcf0e71d5a0d2f2e292c4ed`
- Weth amount (to be provided to exchange for DOT to cover the asset creation fee): `0x00c06e31d91001`
- Token ID (registered asset contract address): `0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2`

Reply:
Thanks Clara. I suspect we're going to be iterating on the registration XCM for a while, and I so I think the unit test:

1. Needs to fleshed out into a fully simulated XCM test
2. The test should automatically generate the byte fragments. See [example code](https://gist.github.com/vgeddes/32237069474ea8f06b35011b5feb11d1) I asked Claude AI to produce.